tensorflow/python/data/ops/dataset_ops.py

```diff
@@ -3479,7 +3489,24 @@ def _to_proto(self):
     pb.optimization_options.CopyFrom(self.experimental_optimization._to_proto())  # pylint: disable=protected-access
     if self.experimental_slack is not None:
       pb.slack = self.experimental_slack
-    pb.threading_options.CopyFrom(self.experimental_threading._to_proto())  # pylint: disable=protected-access
+    # (kvignesh1420): We try to keep the values of `threading` and
+    # `experimental_threading` the same, to prevent unexpected behaviours
+    # and ensure backward-compatibility.
+    if self.threading._has_non_default_values():
+      if self.experimental_threading._has_non_default_values():
+        override_attrs = []
+        for attr in self.threading._get_option_attrs():
+          if (object.__getattribute__(self.threading, attr) !=
+              object.__getattribute__(self.experimental_threading, attr)):
+            override_attrs.append(attr)
+        if override_attrs:
+          logging.warning("overriding attr(s) '{}' of experimental_threading "
+                "with respective values in threading.".format(
+                  ",".join(override_attrs)))
```

Comment:
since we are only checking a small number of values, my suggestion would be to simply check that all attributes of the two objects are identical (instead of checking whether either of them has non-default values)

Reply:
addressed in the commit